app/Watch.hs

```diff
@@ -30,29 +30,35 @@ watchElmFiles flags chan handleEvent = do
 addWatchesRecursively :: FS.Inotify -> FilePath -> IO ()
 addWatchesRecursively inotify dirpath = do
     _ <- FS.addWatch inotify dirpath FS.in_CLOSE_WRITE
-    subdirs <- D.getDirFiltered shouldRecurse dirpath
+    subdirs <- Recursive.listAccessible fsRecurseConf dirpath
     void <| for subdirs <| \subdir -> FS.addWatch inotify subdir FS.in_CLOSE_WRITE
+  where
+    fsRecurseConf :: Recursive.Conf
+    fsRecurseConf =
+        Recursive.Conf
+            { Recursive.preCheck = \path -> not <| any (path `isSuffixOf`) ignoredDirs
```

Comment:
replace this line with

```haskell
            { Recursive.preCheck = \_ -> True
```

if you want to test the performance when recursing into node_modules